### ISSUE TITLE
Click-forwarding into Java applications not working

### ISSUE THREAD

**Issue by ExoMemphiz** [LorenzCK/OnTopReplica#34]
Hi,

I've attempted to make OnTopReplica click-forward to Java, but couldn't get it to work, and was wondering if it's because of some weird interactions it could have with the internals of the Java program (not Java itself, or the JVM, etc.), so I tested it out with the simplest, ugliest GUI I could make in a minute or two:

https://pastebin.com/n8epFtGW

(Basically, a button, that increments the "count" variable, and a label to show what value is in "count")

This did not work, and I was wondering if there is a solution to this, or it's just tough luck? And is there any other alternative (couldn't find anything near the extent of OnTopReplica, and its workings (specifically Click-forwarding based))?

Best Regards

**Comment by ragsmaroon** [comment]
Yeah I tried using it on the Old School Runescape client, and third party clients for the game and they don't seem to work. If I right click, the right click menu comes up in the very top left corner, but other than that no other clicks go through. Super disappointing.

**Comment by carpediembr** [comment]
I've tried using on Runescape 3 (HTML5 Aplication) and the clicks are very imprecise.

**Comment by blundercode** [comment]
Same issue here any solutions yet?

**Comment by AshotN** [comment]
Still an issue

**Comment by MasterCubz** [comment]
Has anyone located a work around for this? I plan to test out running a VM and having the clone be of the VM screen. I'm not familiar with Java, @LorenzCK  is this even possible with the current framework?

**Comment by LorenzCK** [comment]
Great if that works! The problem with cloning apps running within a VM is that OnTopReplica forwards inputs to the VM window or TeamViewer, which in turn forwards them to the target application. So, if this works, it doesnʼt necessarly imply that OnTopReplica works with any Java application running locally (not within a VM). It is my understanding that it mostly depends on the specific application and/or the UI framework. But this is encouraging! :smile:

**Comment by blundercode** [comment]
> Has anyone located a work around for this? I plan to test out running a VM and having the clone be of the VM screen. I'm not familiar with Java, @LorenzCK is this even possible with the current framework?
> 
> Update:
> Functionality works through VM/Teamviewer (Teamviewer has some issue with left clicking but is viable for osrs)

I was also able to play OSRS through TeamViewer, it is a little wonky but works enough.

**Comment by SilverlightPony** [comment]
Trying to use this app with the Nintendo DS emulator DeSmuME, as a way to separate the emulated DS screens onto separate monitors.  Was hoping click-forwarding would let me use the OTR mirror of the lower half work with touch inputs, but clicks aren't being forwarded to the emulator.
